pyoutline/outline/backend/cue.py

```diff
@@ -319,12 +319,15 @@ def _serialize(launcher, use_pycuerun):
 
         # opencue specific options
         # Keeping 'threads' for backward compatibility
-        if layer.get_arg("cores"):
-            if layer.get_arg("threads"):
+        cores = None
+        if layer.get_arg("overrideCores"):
+            logger.info("%s is set to override service cores.", layer.get_name())
+            if layer.is_arg_set("cores") and layer.is_arg_set("threads"):
                 logger.warning("%s has both cores and threads. Use cores.", layer.get_name())
-            sub_element(spec_layer, "cores", "%0.1f" % (layer.get_arg("cores")))
-        elif layer.get_arg("threads"):
-            sub_element(spec_layer, "cores", "%0.1f" % (layer.get_arg("threads")))
+            cores = "%0.1f" % float(layer.get_arg("cores") or layer.get_arg("threads", 0))
+            sub_element(spec_layer, "cores", cores)
+        else:
+            logger.warning("%s will use service cores.", layer.get_name())
```

Comment:
Looking back at this code, it introduces a mandatory argument "overrideCores" that will disrupt existing implementations relying only on pycue.
Another approach could be to test if "cores" is not None, meaning we explicitly override the job's cores value over the service one, thus making current codebases retro-compatible with this feature.
If "cores" is None, then we let the selected service handle that value.
What do you think @DiegoTavares @bcipriano @lithorus ?

Reply:
The change is done and tested.
`overrideCores` stays in cuesubmit.